app/assets/javascripts/discourse/app/components/sidebar/user/categories-section.js

```diff
@@ -1,28 +1,53 @@
 import { inject as service } from "@ember/service";
 import { action } from "@ember/object";
-import Category from "discourse/models/category";
 import { cached } from "@glimmer/tracking";
 
+import Category from "discourse/models/category";
 import SidebarCommonCategoriesSection from "discourse/components/sidebar/common/categories-section";
+import discourseDebounce from "discourse-common/lib/debounce";
+
+export const REFRESH_COUNTS_APP_EVENT_NAME =
+  "sidebar:refresh-categories-section-counts";
 
 export default class SidebarUserCategoriesSection extends SidebarCommonCategoriesSection {
   @service router;
   @service currentUser;
+  @service appEvents;
 
   constructor() {
     super(...arguments);
 
     this.callbackId = this.topicTrackingState.onStateChange(() => {
-      this.sectionLinks.forEach((sectionLink) => {
-        sectionLink.refreshCounts();
-      });
+      this.#refreshCounts();
     });
+
+    this.appEvents.on(REFRESH_COUNTS_APP_EVENT_NAME, this, this.#refreshCounts);
   }
 
   willDestroy() {
     super.willDestroy(...arguments);
 
     this.topicTrackingState.offStateChange(this.callbackId);
+
+    this.appEvents.off(
+      REFRESH_COUNTS_APP_EVENT_NAME,
+      this,
+      this.#refreshCounts
+    );
+  }
+
+  #refreshCounts() {
+    // TopicTrackingState changes or plugins can trigger this function so we debounce to ensure we're not refreshing
+    // unnecessarily.
+    discourseDebounce(
+      this,
+      () => {
+        this.sectionLinks.forEach((sectionLink) => {
+          sectionLink.refreshCounts();
+        });
+      },
```

Comment:
IIRC we can't pass an anonymous function to debounce like this because debounce needs to be called with the same reference to the function for it to work correctly.

Also, maybe we can use our `@debounce` decorator like we do here?:

https://github.com/discourse/discourse/blob/42cf32169d254b3be54cd292b7a78ccf8d56148c/app/assets/javascripts/discourse/app/components/topic-timeline/container.js#L211-L215

Reply:
Good point I've made the changes :+1: